funnel/models/video.py

```diff
@@ -15,6 +15,84 @@ class VideoException(Exception):
     pass
 
 
+def parse_video_url(video_url):
+    video_source = 'raw'
+    video_id = video_url
+
+    parsed = urllib.parse.urlparse(video_url)
+    if parsed.netloc is None:
+        raise ValueError("Invalid video URL")
+
+    if parsed.netloc in ['youtube.com', 'www.youtube.com', 'm.youtube.com']:
+        if parsed.path == '/watch':
+            queries = urllib.parse.parse_qs(parsed.query)
```

Comment:
Nitpick: this is conventionally `args` rather than `queries`. Not a serious issue though.